File: app/src/main/java/com/taitsmith/busboy/data/Bus.java
package com.taitsmith.busboy.data;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Bus  {
    @SerializedName("VehicleId")
    @Expose
    public Integer vehicleId;
    @SerializedName("CurrentTripId")
    @Expose
    public Integer currentTripId;
    @SerializedName("Latitude")
    @Expose
    public Double latitude;
    @SerializedName("Longitude")
    @Expose
    public Double longitude;
    @SerializedName("Heading")
    @Expose
    public Integer heading;
    @SerializedName("TimeLastReported")
    @Expose
    public String timeLastReported;

    public Double getLatitude() {
        return latitude;
    }

    public Double getLongitude() {
        return longitude;
    }
}

File: app/src/main/java/com/taitsmith/busboy/data/Bus.kt
package com.taitsmith.busboy.data

import com.google.gson.annotations.SerializedName
import java.io.Serializable

class Bus: Serializable {
    @SerializedName("VehicleId")
    var vehicleId: Int? = null

    @SerializedName("CurrentTripId")
    var currentTripId: Int? = null

    @SerializedName("Latitude")
    var latitude: Double? = null

    @SerializedName("Longitude")
    var longitude: Double? = null

    @SerializedName("Heading")
    var heading: Int? = null

    @SerializedName("TimeLastReported")
    var timeLastReported: String? = null

    @SerializedName("IsActive")
    var isActive: Boolean? = null

    @SerializedName("Description")
    var description: String? = null

    @SerializedName("VehicleType")
    var vehicleType: String? = null

    @SerializedName("VehicleTypeDescription")
    var vehicleTypeDescription: String? = null

    @SerializedName("Make")
    var make: String? = null

    @SerializedName("SerialNumber")
    var serialNumber: String? = null

    @SerializedName("LicenseNumber")
    var licenseNumber: String? = null

    @SerializedName("Length")
    var length: String? = null

    @SerializedName("PropulsionType")
    var propulsionType: String? = null

    @SerializedName("HasWiFi")
    var hasWiFi: Boolean? = null

    @SerializedName("HasAC")
    var hasAC: Boolean? = null

    @SerializedName("StandingCapacity")
    var standingCapacity: String? = null

    @SerializedName("SeatingCapacity")
    var seatingCapacity: String? = null

    @SerializedName("LimitCapacity")
    var limitCapacity: String? = null

    //so we can save this to a db and keep track of all the buses we've ridden
    var hasRidden: Boolean? = false
}